04-onion-routing.md

```diff
@@ -402,7 +402,7 @@ The construction returns a single 1366-byte packet along with the first receivin
 The packet construction is performed in the reverse order of the route, i.e.
 the last hop's operations are applied first.
 
-The packet is initialized with 1366 `0x00`-bytes.
+The packet is initialized with 1366 _random_ bytes derived from a CSPRNG.
```

Comment:
this doesn't match the code below. the mix header is only 1300 bytes, of which we initialize 1300-firstHopSize bytes via the CSPRNG. since we copy the hop payloads directly into the mix header, we _can_ specify that all 1300 should be random, and leave it as a minor optimization for implementations to not randomize the firstHopSize bytes.

the overall size is 1366 bytes including the session key, next mac, and version, but this is suggesting that those should be initialized via the CSPRNG as well.

Reply:
Yes, in our code we generated the 1300 byte stream using the session key and the 3-byte string "pad" (I know, not a greek letter, but meh).

Reply:
Updated to just fill the set of bytes directly.